cmake/toolchain/iar/Kconfig

```diff
@@ -9,7 +9,28 @@ choice LINKER_SCRIPT
 	default CMAKE_LINKER_GENERATOR
 endchoice
 
-menu "IAR library options"
+menu "IAR options"
+
+choice IAR_INIT
+	bool "Static initialization method"
+	depends on XIP
+	default IAR_ZEPHYR_INIT
+
+config IAR_DATA_INIT
+	bool "IAR handles initialization of static variables"
+	select SKIP_BSS_CLEAR # IAR also handles zero initialization?
+	help
+	  Instead of `z_prep_c` calling Zephyrs `z_data_copy`
+	  we call IARs own proprietary initialization method
+	  which can save time and space.
+
+config IAR_ZEPHYR_INIT
+	bool "Zephyr handles initialization of static variables"
+	help
+	  The traditional `z_data_copy`
```

Comment:
This is a choice with values, and thus look like this:
![image](https://github.com/user-attachments/assets/e4ab90d0-14ab-4f79-befd-6117804dceba)

![image](https://github.com/user-attachments/assets/0d2c4329-bf0d-4f0c-88c8-fda2fa057065)

Often better to shorten the choice entry names, so it looks like this:
![image](https://github.com/user-attachments/assets/6a6a8940-ac0d-4b38-86c4-ed59b69dd0dd)

![image](https://github.com/user-attachments/assets/66c6c7e9-5df5-4f0d-b720-e16f83b24bed)